include/nn/idb/idb_cpp.h

```diff
@@ -0,0 +1,56 @@
+#pragma once
+
+#ifdef __cplusplus
+
+#include <wut.h>
+#include <cstdint>
+
+namespace nn::idb {
+    struct WUT_PACKED FileHeaderEntry {
+        uint64_t titleId;
+        uint32_t u1;
+        WUT_UNKNOWN_BYTES(0xC);
+    };
+    WUT_CHECK_SIZE(FileHeaderEntry, 0x18);
+    WUT_CHECK_OFFSET(FileHeaderEntry, 0x00, titleId);
+    WUT_CHECK_OFFSET(FileHeaderEntry, 0x08, u1);
+
+    struct WUT_PACKED FileHeader {
+        uint32_t version;
+        FileHeaderEntry entries[300];
+    };
+    WUT_CHECK_SIZE(FileHeader, 0x1c24);
+    WUT_CHECK_OFFSET(FileHeader, 0x00, version);
+    WUT_CHECK_OFFSET(FileHeader, 0x04, entries);
+
+    struct WUT_PACKED IconDataKey {
+        uint64_t titleId;
+        uint32_t u1; // maybe some kind of filter (or version)?
+    };
+    WUT_CHECK_SIZE(IconDataKey, 0xC);
+    WUT_CHECK_OFFSET(IconDataKey, 0x00, titleId);
+    WUT_CHECK_OFFSET(IconDataKey, 0x08, u1);
+
+    struct WUT_PACKED Utf16NameBuffer {
+        char utf16Str[0x200];
```

Comment:
Any reason for not using `uint16_t utf16Str[0x100];` here if it's UTF16?

Reply:
This stuff is C++ so why not use `char16_t?[0x100];`

Reply:
done